avro4s-core/src/main/scala/com/sksamuel/avro4s/Encoder.scala

```diff
@@ -121,6 +122,45 @@ object Encoder
     }
   }
 
+  case class EncoderField[T](name: String, schema: Schema, extractor: T => Any)
+
+  object EncoderField {
+    def string[T](name: String, extractor: T => Any) = EncoderField(name, Schema.create(Schema.Type.STRING), extractor)
```

Comment:
shouldn't it be
```scala
def field[T, F](name: String, extractor: T => F)(implicit encoder: Encoder[F]): EncoderField[T, F] = ...
```
?

Reply:
this would make it possible to use the schema from the `encoder` too

Reply:
Done.